fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectFactory.java

```diff
@@ -54,203 +48,167 @@
 
     private FcrepoClient client;
 
+    private RepositoryPaths repoPaths;
+
+    private RepositoryObjectDataLoader repoObjDataLoader;
+
+    private RepositoryObjectLoader repoObjLoader;
+
+    private RepositoryPIDMinter pidMinter;
+
     /**
-     * Creates a deposit record object structure at the given path with the
-     * properties specified in the provided model
+     * Creates a new deposit record object with the given uuid.
+     * Properties in the supplied model will be added to the deposit record.
      *
-     * @param path
-     *            URI of the full path where this deposit record should be
-     *            created
+     * @param pid
      * @param model
-     *            Model containing additional properties to add to this Deposit
-     *            Record. Optional.
-     * @return the URI of the created deposit record
+     * @return
      * @throws FedoraException
      */
-    public URI createDepositRecord(URI path, Model model) throws FedoraException {
-        // Add the deposit record type to the object being created
-        model = populateModelTypes(path, model, Arrays.asList(Cdr.DepositRecord));
+    public DepositRecord createDepositRecord(PID pid, Model model) throws FedoraException {
```

Comment:
The signatures here are good, but I am wondering where the work of actually creating the object in fedora is moving to?  It could be directly in these methods or could still be in another class.  But someones got to do the dirty work in the fedora api!